### PR TITLE
Replacing more C-style for loops with the faster, traditional ones.

### PR DESCRIPTION
**Pull request by Ghommie**
## About The Pull Request
Continuation of #62624.

## Why It's Good For The Game
Making the code more readable and less expensive.

## Changelog
N/A

### CODE REVIEW

File: code/__HELPERS/_lists.dm [hunk 1]
```diff
@@ -358,7 +358,7 @@
 		return
 	inserted_list = inserted_list.Copy()
 
-	for(var/i = 1, i < inserted_list.len, ++i)
+	for(var/i in 1 to inserted_list.len)
```
Comment (Mothblocks) [inline code/__HELPERS/_lists.dm:361]:
```suggestion
	for(var/i in 1 to inserted_list.len - 1)
```

You're changing behavior.

File: code/__HELPERS/_lists.dm [hunk 2]
```diff
@@ -368,7 +368,7 @@
 	if(!inserted_list)
 		return
 
-	for(var/i = 1, i < inserted_list.len, ++i)
+	for(var/i in 1 to inserted_list.len)
```
Comment (Mothblocks) [inline code/__HELPERS/_lists.dm:371]:
```suggestion
	for(var/i in 1 to inserted_list.len - 1)
```

File: code/__HELPERS/_lists.dm (hunk 1 above)
Comment (Mothblocks) [reply code/__HELPERS/_lists.dm:361]: image

File: code/__HELPERS/_lists.dm [hunk 3]
```diff
@@ -472,15 +472,15 @@
 			return //no need to move
 		from_index += len //we want to shift left instead of right
 
-		for(var/i = 0, i < distance, ++i)
+		for(var/i in 1 to distance)
```
Comment (Mothblocks) [inline code/__HELPERS/_lists.dm:475]: This is wrong in the same vein, but I wonder if the old code was just always wrong?

File: code/__HELPERS/_lists.dm [hunk 4]
```diff
@@ -472,15 +472,15 @@
...
 			inserted_list.Insert(from_index, null)
 			inserted_list.Swap(from_index, to_index)
 			inserted_list.Cut(to_index, to_index + 1)
 	else
 		if(from_index > to_index)
 			from_index += len
 
-		for(var/i = 0, i < len, ++i)
+		for(var/i in 1 to len)
```
Comment (Mothblocks) [inline code/__HELPERS/_lists.dm:483]: Same

File: code/__HELPERS/_lists.dm [hunk 5]
```diff
@@ -506,7 +506,7 @@
 			to_index = from_index
 			from_index = a
 
-		for(var/i = 0, i < len, ++i)
+		for(var/i in 1 to len)
```
Comment (Mothblocks) [inline code/__HELPERS/_lists.dm:509]: Same

File: code/__HELPERS/_lists.dm [hunk 6]
```diff
@@ -496,7 +496,7 @@
 		else
 			from_index += len
 
-		for(var/i = 0, i < distance, ++i)
+		for(var/i in 1 to distance)
```
Comment (Mothblocks) [inline code/__HELPERS/_lists.dm:499]: Same

**Review by Mothblocks — CHANGES_REQUESTED** [review]

File: code/__HELPERS/_lists.dm (hunk 3 above)
Comment (Ghommie) [reply code/__HELPERS/_lists.dm:475]: This one is not wrong. `i` starts at 0 here.

File: code/__HELPERS/_lists.dm (hunk 4 above)
Comment (Ghommie) [reply code/__HELPERS/_lists.dm:483]: See above.

File: code/__HELPERS/_lists.dm (hunk 6 above)
Comment (Ghommie) [reply code/__HELPERS/_lists.dm:499]: See above.

**Comment by Mothblocks** [comment]
@Ghommie Yes, it is wrong. The reason I commented is because it looks like list operations, so I can't tell if the old code was just bugged.

image

**Comment by Mothblocks** [comment]
Nevermind, I see now, `i` wasn't used.

**Pull request merged**